test/battle/ability/commander.c

```diff
@@ -0,0 +1,440 @@
+            ABILITY_POPUP(playerLeft, ABILITY_COMMANDER);
+            MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        }
+    }
+}
+
+DOUBLE_BATTLE_TEST("Order Up increases a stat based on Tatsugiri's form")
+{
+    u32 species = 0;
+    PARAMETRIZE { species = SPECIES_TATSUGIRI_CURLY; }
+    PARAMETRIZE { species = SPECIES_TATSUGIRI_DROOPY; }
+    PARAMETRIZE { species = SPECIES_TATSUGIRI_STRETCHY; }
+
+    GIVEN {
+        PLAYER(species) { Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_DONDOZO);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_VOLBEAT) { Ability(ABILITY_PRANKSTER); };
+    } WHEN {
+        TURN { MOVE(opponentRight, MOVE_HAZE); MOVE(playerRight, MOVE_ORDER_UP, target: opponentLeft); }
+    } SCENE {
+        ABILITY_POPUP(playerLeft, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, playerRight);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_HAZE, opponentRight); // Remove previous stat boosts
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_ORDER_UP, playerRight);
+        switch (species)
+        {
+        case SPECIES_TATSUGIRI_CURLY:
+            MESSAGE("Dondozo's Attack rose!");
+            break;
+        case SPECIES_TATSUGIRI_DROOPY:
+            MESSAGE("Dondozo's Defense rose!");
+            break;
+        case SPECIES_TATSUGIRI_STRETCHY:
+            MESSAGE("Dondozo's Speed rose!");
+            break;
+        }
+    } THEN {
+        switch (species)
+        {
+        case SPECIES_TATSUGIRI_CURLY:
+            EXPECT_EQ(playerRight->statStages[STAT_ATK], DEFAULT_STAT_STAGE + 1);
+            break;
+        case SPECIES_TATSUGIRI_DROOPY:
+            EXPECT_EQ(playerRight->statStages[STAT_DEF], DEFAULT_STAT_STAGE + 1);
+            break;
+        case SPECIES_TATSUGIRI_STRETCHY:
+            EXPECT_EQ(playerRight->statStages[STAT_SPEED], DEFAULT_STAT_STAGE + 1);
+            break;
+        }
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Whirlwind can not be used against Dondozo or Tatsugiri while Commander is active")
+{
+    GIVEN {
+        PLAYER(SPECIES_TATSUGIRI) { Ability(ABILITY_COMMANDER); }
+        PLAYER(SPECIES_DONDOZO);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponentLeft, MOVE_WHIRLWIND, target: playerLeft); }
+        TURN { MOVE(opponentRight, MOVE_WHIRLWIND, target: playerRight); }
+    } SCENE {
+        ABILITY_POPUP(playerLeft, ABILITY_COMMANDER);
+        MESSAGE("Tatsugiri was swallowed by Dondozo and became Dondozo's commander!");
+        MESSAGE("Foe Wobbuffet used Whirlwind!");
+        MESSAGE("But it failed!");
+        MESSAGE("Foe Wobbuffet used Whirlwind!");
+        MESSAGE("But it failed!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("(Commander) Red Card fails on Dondozo while Commander is active")
```

Comment:
I'm not sure if I'm fully understanding this test, but if it's supposed to use Red Card but not switch out, `BattleScript_RedCardActivates` probably needs a case for that, also might be useful to check if `playerRight`'s species is still Dondozo and if the Red Card holder has an item at the end.

Reply:
oh right, forgot to address this one. Will do later

Reply:
Wait this test makes no sense, my bad lol

But generally yes. If red card is triggered on dozo it isn't switched out